### PR TITLE
[Docs] Add arcsight module docs contributed by acchen97

### PR DESCRIPTION
**Pull request by dedemorton**
@acchen97  I've had to make some minor formatting changes to conform to Elastic style, but did not change the language unless otherwise noted in a comment. I don't have time to put these up on firebase, so you can either pull the files and run your own local build, or just review the content and wait until it's published and take a look at the formatting.

Please note that I did not test the steps because this content has already been tested and reviewed by Alvin and others.

TODO:
- [ ] After I merge the content about netflows, I need to use the shared file for common options rather than duplicating them.

### CODE REVIEW

File: docs/static/arcsight-module.asciidoc [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+module name, i.e. `arcsight.var.inputs` instead of `var.inputs`.
+
+All settings are optional. If you don't specify configuration settings, Logstash
+uses the defaults.
+
+//ALVIN: I don't see a good reason to have the summary table plus the reference info, so I have not included the table. It doesn't add enough value to warrant the real estate it takes up. Plus...tables don't work well when we go from docbook > PDF. If you feel strongly about having it, you can submit a separate PR, but make sure you make each setting an active link, or users will be extra annoyed by the repeated info.
+
+*`var.inputs`*::
++
+--
```
Comment (dedemorton) [inline docs/static/arcsight-module.asciidoc]: The formatting might look a bit unusual. The dashes are needed to include bulleted lists within definition lists.

(DLs are our standard way to format config options...not used consistently across the docs, but the recommended way.)

File: docs/static/arcsight-module.asciidoc [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+image::static/images/arcsight-architecture-diagram-2017.png[ArcSight architecture]
+
+[[arcsight-architecture]]
+==== Deployment Architecture
+
+The Logstash ArcSight module understands CEF (Common Event Format), and can
+accept, enrich, and index these events for analysis on the Elastic Stack. ADP
+contains two core data collection components for data streaming:
+
+* The _EventBroker (EB)_ is the central hub for incoming data and is based on
```
Comment (acchen97) [inline docs/static/arcsight-module.asciidoc]: I should have changed this earlier, but lets add spaces here. "Event Broker" and "Smart Connectors"

File: docs/static/arcsight-module.asciidoc [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+===== Logstash ArcSight Module Configuration Options
+
+These are the configurable settings available for the Logstash ArcSight module.
+When overriding settings in the command line, the setting option must be
+prefixed with the module name, i.e. `arcsight.var.inputs` instead of `var.inputs`.
+
+All settings are optional. If you don't specify configuration settings, Logstash
+uses the defaults.
+
+//ALVIN: I don't see a good reason to have the summary table plus the reference info, so I have not included the table. It doesn't add enough value to warrant the real estate it takes up. Plus...tables don't work well when we go from docbook > PDF. If you feel strongly about having it, you can submit a separate PR, but make sure you make each setting an active link, or users will be extra annoyed by the repeated info. I've modified the intro text slightly to reflect this change.
```
Comment (acchen97) [inline docs/static/arcsight-module.asciidoc]: fine by me! :+1:

**Comment by dedemorton** [comment]
noting that I forgot to add the images. I'll add them later tonight

**Comment by acchen97** [comment]
Ah good catch, thanks @dedemorton

**Pull request closed without merging**